wallet-web/components/bond/BondNodeForm.tsx

```diff
@@ -1,380 +1,480 @@
-import TextField from '@material-ui/core/TextField';
-import { Button, Checkbox, FormControlLabel, InputAdornment } from "@material-ui/core";
-import bs58 from "bs58";
-import semver from "semver"
-import { NodeType } from "../../common/node";
-import { theme } from "../../lib/theme";
-import { basicRawCoinValueValidation, makeBasicStyle, validateRawPort } from "../../common/helpers";
-import { Coin, nativeToPrintable } from "@nymproject/nym-validator-client";
-import { DENOM } from "../../pages/_app";
-import { printableBalanceToNative } from "@nymproject/nym-validator-client/dist/currency";
-import { BondingInformation } from "./NodeBond";
+import React, { useEffect, useState, ChangeEvent } from 'react'
+import { printableBalanceToNative } from '@nymproject/nym-validator-client/dist/currency'
+import { Coin, nativeToPrintable } from '@nymproject/nym-validator-client'
+import { Alert } from '@material-ui/lab'
+import Grid from '@material-ui/core/Grid'
+import TextField from '@material-ui/core/TextField'
+import {
+  Button,
+  Checkbox,
+  FormControlLabel,
+  InputAdornment,
+} from '@material-ui/core'
+import bs58 from 'bs58'
+import semver from 'semver'
+import { NodeType } from '../../common/node'
+import { theme } from '../../lib/theme'
+import {
+  basicRawCoinValueValidation,
+  makeBasicStyle,
+  validateRawPort,
+} from '../../common/helpers'
+import { DENOM } from '../../pages/_app'
+import { BondingInformation } from './NodeBond'
+import { useGetBalance } from '../../hooks/useGetBalance'
 
 const DEFAULT_MIX_PORT = 1789
 const DEFAULT_VERLOC_PORT = 1790
 const DEFAULT_HTTP_API_PORT = 8000
 const DEFAULT_CLIENTS_PORT = 9000
 
-type BondNodeFormProps = {
-    type: NodeType
-    minimumMixnodeBond: Coin,
-    minimumGatewayBond: Coin,
-    onSubmit: (event: any) => void
+type TBondNodeFormProps = {
+  type: NodeType
+  minimumMixnodeBond: Coin
+  minimumGatewayBond: Coin
+  onSubmit: (event: any) => void
 }
 
-export default function BondNodeForm(props: BondNodeFormProps) {
-    const classes = makeBasicStyle(theme);
-
-    const [validity, setValidity] = React.useState({
-        validAmount: true,
-        validSphinxKey: true,
-        validIdentityKey: true,
-        validHost: true,
-        validVersion: true,
-        validLocation: true,
-        validMixPort: true,
-
-        // this should have probably be somehow split to be subclasses of the validity matrix
-        // the above is more true now as more fields are added. This looks kinda disgusting...
-        // mixnode-specific:
-        validVerlocPort: true,
-        validHttpApiPort: true,
-
-        // gateway-specific:
-        validClientsPort: true,
-    })
-
-    const [advancedShown, setAdvancedShown] = React.useState(false)
+type TFormInput = {
+  amount: string
+  host: string
+  http_api_port: string
+  mixPort: string
+  verlocPort: string
+  sphinxKey: string
+  identityKey: string
+  version: string
+  location?: string
+  clientsPort?: string
```

Comment:
Oh. Optional fields. I totally forgot they existed when I first wrote it

Reply:
:smiley: I thought a type for the form input to include these values would help us to see what the form is dealing with.